### PR TITLE
Issue smaller query when checking out buffer from queue

### PR DESCRIPTION
**Pull request by gravityrail**
Previously some hosts were throwing an error because the query to check out queue items with size was greater than 16k in size. This is because it was using an IN query to select rows by ID. This PR changes the query to use a range between a maximum ID and a minimum ID.

### CODE REVIEW

File: sync/class.jetpack-sync-queue.php [hunk 1]
```diff
@@ -196,21 +196,36 @@ function checkout_with_memory_limit( $max_memory, $max_buffer_size = 500 ) {
...
-			$item_ids[] = $item_with_size->id;
+
+			$max_item_id = $item_with_size->id;
 		}
 
-		$items = $this->fetch_items_by_id( $item_ids );
+		$query = $wpdb->prepare( 
+			"SELECT option_name AS id, option_value AS value FROM $wpdb->options WHERE option_name >= %s and option_name <= %s ORDER BY option_name ASC",
+			$min_item_id,
+			$max_item_id
```
Comment (ebinnion) [inline sync/class.jetpack-sync-queue.php]: Do you expect any issues from using `>` and `<` comparisons with [how we currently name our options](https://github.com/Automattic/jetpack/blob/72caed3f9334e087ff435c315c28095aa32f367a/sync/class.jetpack-sync-queue.php#L343-L359)?

Comment (gravityrail) [inline sync/class.jetpack-sync-queue.php]: I don't expect any issues, as this mirrors the behaviour of the original select that returns the options (i.e. it selects all options by option_name ascending)

**Comment by ebinnion** [comment]
No comments on code. I tested with a full sync and this worked fine for me. 🚢

**Comment by gravityrail** [comment]
Merged to 4.2

**Comment by kraftbj** [comment]
Tested this on WP Engine, all good.

**Pull request merged**